### PR TITLE
Expose "length" filter, make it work with Countable objects

### PR DESCRIPTION
**Pull request by rundef**
I needed to count the number of elements in an array ...
I saw that this feature has been talked about in issue #33

I wasn't sure if I should have created a new filter `|count` for objects only ... Tell me what you think!

### CODE REVIEW

File: src/Latte/Runtime/Filters.php [hunk 1]
```diff
@@ -366,7 +366,9 @@ public static function capitalize($s)
 	 */
 	public static function length($s)
 	{
-		return strlen(utf8_decode($s)); // fastest way
+		return is_string($s) ?
```
Comment (fprochazka) [inline src/Latte/Runtime/Filters.php]: imho it should be something like

```php
public static function length($s)
{
    if (is_array($s)) {
        return count($s);
    } elseif ($s instanceof \Traversable) {
        return iterator_count($s);
    } else {
        return strlen(utf8_decode($s)): // fastest way
    }
```

**Comment by xificurk** [comment]
:+1: 

A test would be nice ;-)

File: src/Latte/Runtime/Filters.php (hunk 1 above)
Comment (dg) [inline src/Latte/Runtime/Filters.php]: if (is_array($val) || $val instanceof \Countable)

**Comment by rundef** [comment]
Thanks for the quick feedback :)

**Comment by dg** [comment]
Thanks, merged

File: src/Latte/Runtime/Filters.php [hunk 2]
```diff
@@ -366,7 +366,13 @@ public static function capitalize($s)
 	 */
 	public static function length($s)
 	{
-		return strlen(utf8_decode($s)); // fastest way
+		if (is_array($s) || $s instanceof \Countable) {
+			return count($s);
+		} elseif ($s instanceof \Traversable) {
+			return iterator_count($s);
```
Comment (hrach) [inline src/Latte/Runtime/Filters.php]: this (may) mutate the iterator, imo it's best to let decide the programmer.

> iterator_count() is not guaranteed to retain the current position of the iterator. 
> http://php.net/iterator_count

**Pull request closed without merging**